File: lib/features/banknote_classifier/domain/banknote_classifier.dart
import 'dart:typed_data';

import '../../../core/errors/result.dart';

/// Абстрактный интерфейс классификатора денежных купюр (§4.7 ТЗ).
///
/// Should-have фича, защищена feature flag.
/// Если [FeatureFlags.banknoteClassifier] == false,
/// Command Processor возвращает «Функция пока недоступна».
abstract class BanknoteClassifier {
  /// Классифицирует купюру на изображении.
  ///
  /// [imageBytes] — байты изображения (crop центра 60% уже выполнен).
  ///
  /// Возвращает:
  /// - [Success<String>] с описанием («500 рублей»)
  /// - [Failure] при низком confidence или ошибке
  Future<Result<String>> classify(Uint8List imageBytes);

  /// Освобождает ресурсы.
  Future<void> dispose();
}

File: src/features/banknoteClassifier/domain/banknoteClassifier.ts
/**
 * Абстрактный интерфейс классификатора купюр.
 */

import { Result } from '@core/errors/result';

export interface BanknoteClassifier {
  /**
   * Классифицировать купюру на изображении.
   * @param imageInput — путь к файлу или готовый RGB-тензор
   * @returns название купюры (напр. «500 рублей») или Failure
   */
  classify(imageInput: string | Uint8Array): Promise<Result<string>>;

  /** Загрузить модель */
  initialize(): Promise<Result<void>>;

  /** Освободить ресурсы */
  dispose(): Promise<void>;

  /** Загружена ли модель */
  isReady(): boolean;
}
